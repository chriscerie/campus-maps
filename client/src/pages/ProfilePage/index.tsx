import { Avatar } from '@mui/material';
import { Grid, Container } from '@mui/material';
import { Fragment } from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../reducers';
import './index.scss';

function ProfilePage() {
  const currentUser = useSelector((state: RootState) => state.currentUser);

  return (
    <div className="profile-page-container align-text-left">
      {currentUser ? (
        <Fragment>
          <Container>
            <div className="header-text">Profile</div>
            <Grid container>
              <Grid item xs={12}>
                <div className="profile-icon-container">
                  <Avatar
                    id="profile-greeting-icon"
                    src={currentUser ? currentUser.photo : ''}
                  />
                  <div id="profile-greeting-text">
                    Hello, {currentUser.name}!
                  </div>
                </div>
              </Grid>
              <Grid item xs={12}>
                <div className="profile-info">
                  <strong>Email: </strong> {currentUser.email}
                </div>
              </Grid>
            </Grid>
          </Container>
        </Fragment>
      ) : (
        <Fragment>
          <Container>
            <div className="header-text">Profile</div>
            <Grid container>
              <Grid item xs={12}>
                <div className="profile-icon-container">
                  <Avatar id="profile-greeting-icon" src={''} />
                  <div id="profile-greeting-text">You are not logged in!</div>
                </div>
              </Grid>
            </Grid>
          </Container>
        </Fragment>
      )}
    </div>
  );
}

export default ProfilePage;
