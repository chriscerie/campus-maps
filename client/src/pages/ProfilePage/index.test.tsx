import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../reducers';
import ProfilePage from './index';

describe('ProfilePage', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(
      <Provider store={store}>
        <ProfilePage />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
