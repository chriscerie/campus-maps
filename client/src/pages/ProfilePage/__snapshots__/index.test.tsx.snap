// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage should render successfully 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ProfilePage />
</Context.Provider>
`;
